test/actions/parse_csv_test.exs

```diff
@@ -0,0 +1,172 @@
+defmodule WorkflowEngine.Actions.ParseCsvTest do
+  use ExUnit.Case, async: true
+  use OK.Pipe
+
+  import Mox
+
+  setup :verify_on_exit!
+
+  describe "CSV Action" do
+    test "Parse CSV with comma" do
+      {:ok, result} =
+        build_workflow(",")
+        |> WorkflowEngine.evaluate(
+          params: %{
+            "csv_data" => "name,age\nJohn,30\nJane,25"
+          }
+        )
+        ~> WorkflowEngine.State.get_var("content_rows")
+        ~> Enum.to_list()
+
+      assert result == [
+               %{"name" => "John", "age" => "30"},
+               %{"name" => "Jane", "age" => "25"}
+             ]
+    end
+
+    test "Parse CSV with semicolon" do
+      {:ok, result} =
+        build_workflow(";")
+        |> WorkflowEngine.evaluate(
+          params: %{
+            "csv_data" => "name;age\nJohn;30\nJane;25"
+          }
+        )
+        ~> WorkflowEngine.State.get_var("content_rows")
+        ~> Enum.to_list()
+
+      assert result == [
+               %{"name" => "John", "age" => "30"},
+               %{"name" => "Jane", "age" => "25"}
+             ]
+    end
+
+    test "Parse CSV with tab" do
+      {:ok, result} =
+        build_workflow("\t")
+        |> WorkflowEngine.evaluate(
+          params: %{
+            "csv_data" => "name\tage\nJohn\t30\nJane\t25"
+          }
+        )
+        ~> WorkflowEngine.State.get_var("content_rows")
+        ~> Enum.to_list()
+
+      assert result == [
+               %{"name" => "John", "age" => "30"},
+               %{"name" => "Jane", "age" => "25"}
+             ]
+    end
+
+    test "download CSV with http action with redirect" do
```

Comment:
Testing the redirect behaviour in this file is not fine as it mixes concerns. Please move redirect true/false tests to the HTTP action test module.
Also, we can't use a public file in our automated test suite. I don't think that we need to explicitly test downloading a file and then parsing it. Reading a value from the workflow context (using `data: { var: "…" }`) is enough.